### ISSUE TITLE
[BUG] npm ci fails if node_modules exist

### ISSUE THREAD

**Issue by mildaniel** [npm/cli#6056]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [x] I am using the latest npm

### Current Behavior

When running `npm ci` in a directory with an existing `node_modules` that contains a `.bin` directory, the command fails with the following exception:
```
43 verbose stack SystemError [ERR_FS_EISDIR]: Path is a directory: rm returned EISDIR (is a directory) {PATH_TO_PROJECT}/node_modules/.bin
43 verbose stack     at new SystemError (node:internal/errors:250:5)
43 verbose stack     at new NodeError (node:internal/errors:361:7)
43 verbose stack     at node:internal/fs/utils:784:17
43 verbose stack     at callback (/usr/local/lib/node_modules/npm/node_modules/graceful-fs/polyfills.js:306:20)
43 verbose stack     at FSReqCallback.oncomplete (node:fs:191:5)
```

### Expected Behavior

Expect `npm ci` to work regardless of if the `node_modules` directory already exists.

### Steps To Reproduce

1. With this config
Sample `package.json`
```
{
  "name": "esbuild-binary",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "author": "",
  "license": "ISC",
  "dependencies": {
    "esbuild": "^0.14.13"
  }
}

```
2. Generate lockfile then run `npm ci`

### Environment

- npm: 9.3.0
- Node.js: 19.4.0
- OS Name: MacOS Monterey 12.6.1
- System Model Name: Macbook Pro

**Comment by ParadoxInfinite** [comment]
@mildaniel This seems to be fixed in the `latest` branch. I could reproduce your issue on the 9.3.0 release commit https://github.com/npm/cli/commit/4d275927e069606c8047b678ad09aafbe18b1673, however, when I switch to the `latest` branch, which at the time of this response is at https://github.com/npm/cli/commit/8be672b21f7d434c898bb4e97dc3fca6e54b29a1, I am able to `npm ci` without a problem.

So there's no point in changing any code for now. You could wait for the new release, or use an older version, which works for your use case, until then.

**Comment by cyberbiont** [comment]
I have the same problem with npm 9.3.0

**Comment by ernestoalejo** [comment]
This issue is a duplicate of the previous one, already fixed: https://github.com/npm/cli/issues/6051

There isn't a release with the fix yet though. Using an old version like 9.2.0 seems like the best option for now.

**Comment by lukekarrys** [comment]
The fix for this has been released in `v9.3.1`
